integreat_cms/static/src/js/pages/fetch-subpages.ts

```diff
@@ -3,42 +3,64 @@
  */
 
 import { create_icons_at } from "../utils/create-icons";
+import { getCsrfToken } from "../utils/csrf-token";
 import { setBulkActionEventListeners } from "../bulk-actions";
 import { setToggleSubpagesEventListeners } from "./toggle-subpages";
 import { addDragAndDropListeners } from "../tree-drag-and-drop";
 import { addConfirmationDialogListeners } from "../confirmation-popups";
 import { addPreviewWindowListeners, openPreviewWindowInPageTree } from "./page-preview";
+import { restorePageTreeLayout } from "./persistent_page_tree";
 
-window.addEventListener("load", () => {
+window.addEventListener("load", async () => {
   // Load subpages initially
-  if (document.querySelector("[data-delay-event-handlers]")) {
-    let rootPages: HTMLElement[] = Array.from(document.querySelectorAll(".toggle-subpages"));
+  const table: HTMLElement = document.querySelector("[data-delay-event-handlers]");
+  if (table) {
+    const rootPages: HTMLElement[] = Array.from(document.querySelectorAll(".toggle-subpages"));
+    const url = table.dataset.descendantsUrl;
     console.debug("Loading all subpages");
-    Promise.all(rootPages.map(fetchSubpages)).then(() => {
-      console.debug("Finished loading all subpages");
-      // Set event handlers
-      setToggleSubpagesEventListeners();
-      setBulkActionEventListeners();
-      addDragAndDropListeners();
-      addConfirmationDialogListeners();
-      addPreviewWindowListeners(openPreviewWindowInPageTree);
-    })
+
+    let expandedRows: Set<number> = new Set(JSON.parse(window.sessionStorage.getItem("page_state")));
+    const openPages = rootPages.filter(page => expandedRows.has(Number(page.dataset.pageId)));
+    const closedPages = rootPages.filter(page => !expandedRows.has(Number(page.dataset.pageId)));
+
+    await fetchAllSubpages(url, openPages);
+    restorePageTreeLayout();
+
+    await fetchAllSubpages(url, closedPages);
```

Comment:
Very cool idea :ok_hand:

Reply:
Thanks!

Reply:
Ok, so the only issue I see with it is that it takes longer than before when no page was expanded or all pages were. Do you think it would be feasible to keep the previous solution of fetching each tree individually and just perform them in two stages?
E.g. like this?
```
await Promise.all(openPages.map(fetchSubpages))
restorePageTreeLayout();
await Promise.all(closedPages.map(fetchSubpages))
```

Reply:
So if I understand correctly, do you want to go back to sending multiple requests at once? This does not seem to be faster for me, at least locally

Reply:
I think this depends a lot. Our server definitely has multiple processes running, which is a difference to local test environments. I think it is very valid to send multiple requests at once.

Reply:
I set up a multi-process-environment locally and checked whether the alternative fetching approach was really faster, but it wasn't. Could be my specific setup, but I don't think it's worth the effort testing this on the real server, probably the current solution is fine. Feeling a little bit ashamed that my objection caused this PR to be open for nearly a year :see_no_evil: :crying_cat_face: